server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/WriteLoadConstraintDeciderIT.java

```diff
@@ -451,10 +819,12 @@ private Settings enabledWriteLoadDeciderSettings(int utilizationThresholdPercent
                 WriteLoadConstraintSettings.WRITE_LOAD_DECIDER_HIGH_UTILIZATION_THRESHOLD_SETTING.getKey(),
                 utilizationThresholdPercent + "%"
             )
-            // TODO (ES-12862): remove these overrides when throttling is turned off for simulations.
-            .put(CLUSTER_ROUTING_ALLOCATION_NODE_CONCURRENT_INCOMING_RECOVERIES_SETTING.getKey(), 100)
-            .put(CLUSTER_ROUTING_ALLOCATION_NODE_CONCURRENT_OUTGOING_RECOVERIES_SETTING.getKey(), 100)
-            .put(CLUSTER_ROUTING_ALLOCATION_NODE_CONCURRENT_RECOVERIES_SETTING.getKey(), 100)
```

Comment:
Since Yang added the logic to return after every shard movement, the balancer won't hit THROTTLE anymore.

I initially had to add a flag to do either 100 or 1, so the new canRemain logic could set 1 and return from the BalancedShardsAllocator to simulate each shard move in the DesiredBalanceShardsAllocator before making another decision. Otherwise canRemain would keep returning NOT_PREFERRED for every shard until simulation changed it. But now I don't need to manipulate these settings for either case.